### PR TITLE
Fix polyfillLazyGlobal to work with allowTopLevelThis = false

### LINKED ISSUE #7700: babel strict mode transform is not working with v0.26
since v0.26 babel strict mode transform is not working anymore. I get different errors mostly of internal methods which try to use setters on read-only objects.

### PR DESCRIPTION
**Pull request by janicduplessis**
`polyfillLazyGlobal` used a top level this which get stripped by babel `transform-es2015-modules-commonjs` with the default config. This is mainly an issues for people not using the react native babel preset.

This also replaces a few GLOBAL with global for consistency with the rest of the file.

**Test plan**

Tested that there was an error when using `['transform-es2015-modules-commonjs', { strict: true, allowTopLevelThis: false }]` in the babel config and that it was fixed after applying my changes.

Fixes #7700

### CODE REVIEW

**Comment by facebook-github-bot-3** [comment]
By analyzing the blame information on this pull request, we identified @javache and @davidaurelio to be potential reviewers.

File: Libraries/JavaScriptAppEngine/Initialization/InitializeJavaScriptAppEngine.js [hunk 1]
```diff
@@ -33,10 +33,10 @@ if (typeof window === 'undefined') {
 }
 
 function setUpProcess() {
-  GLOBAL.process = GLOBAL.process || {};
-  GLOBAL.process.env = GLOBAL.process.env || {};
-  if (!GLOBAL.process.env.NODE_ENV) {
-    GLOBAL.process.env.NODE_ENV = __DEV__ ? 'development' : 'production';
+  global.process = global.process || {};
```
Comment (javache) [inline Libraries/JavaScriptAppEngine/Initialization/InitializeJavaScriptAppEngine.js]: Can you update the globals definition on line 23 too?

**Comment by javache** [comment]
@davidaurelio These options actually look pretty useful to enable by default.

**Comment by janicduplessis** [comment]
@javache We already tried enabling these by default but it caused a lot of issues since we compile everything including node_module libraries that are not always es module compliant. Last time I checked babel didn't support a way to pass options to presets so as long as there is not an easy way to customize these options I think it's best to use the more loose mode.

**Comment by facebook-github-bot-5** [comment]
@janicduplessis updated the pull request.

**Comment by satya164** [comment]
I add this to my `.babelrc` so that third part libs in `node_modules` are not compiled.

```json
"ignore": "node_modules/!(react-native*|react-clone-referenced-element)*/**"
```

**Comment by javache** [comment]
@facebook-github-bot shipit

**Comment by facebook-github-bot-0** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://our.intern.facebook.com/intern/opensource/github/pull_request/286550338347678/int_phab) to review.

**Pull request closed without merging**